### PR TITLE
Reverting narayana.bat

### PR DESCRIPTION
**Pull request by mayankkunwar**
Reverting the deleted `narayana.bat` file, which was deleted as part of https://github.com/jbosstm/narayana/pull/1873

MAIN CORE !TOMCAT !AS_TESTS !RTS !JACOCO !XTS !QA_JTA !QA_JTS_JACORB !QA_JTS_JDKORB !QA_JTS_OPENJDKORB !PERF !LRA !NO_WIN !DB_TESTS !mysql !db2 !postgres !oracle

### CODE REVIEW

**Review by mmusgrov — CHANGES_REQUESTED** [review]
This batch file contains a number of references to the BlackTie project which no longer exists.

File: scripts/hudson/narayana.bat [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  if %ERRORLEVEL% NEQ 0 exit -1
+  cd ..
+  call build.bat clean install "-f" "jboss-transaction-spi\pom.xml" || (call:comment_on_pull "SPI build Failed %BUILD_URL%" & exit -1)
+goto:eof
+
+:fail_build
+  call:comment_on_pull "Build failed %BUILD_URL%"
+  tasklist & FOR /F "usebackq tokens=5" %%i in (`"netstat -ano|findstr 9990.*LISTENING"`) DO taskkill /F /PID %%i
+  tasklist
+  taskkill /F /IM mspdbsrv.exe
```
Comment (mmusgrov) [inline scripts/hudson/narayana.bat:57]: I am pretty sure these are blacktie specific.

File: scripts/hudson/narayana.bat [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+java -version 2>tmp.txt
+findstr version tmp.txt > version.txt
+FOR /F delims^=^"^ tokens^=2 %%i in (version.txt) DO @set JAVA_VERSION=%%i
+if %JAVA_VERSION% == 9 set MAVEN_OPTS="--add-modules=java.corba,java.xml.bind,java.xml.ws"
+
+IF NOT [%NOTMAIN%] == [] (call:build_spi_pr
+call build.bat clean install %* || (call:comment_on_pull "Tests failed on Windows - Narayana Failed %BUILD_URL%" & exit -1)
+)
+
+IF NOT [%NOTBT%] == [] (echo "set WILDFLY_MASTER_VERSION=%WILDFLY_MASTER_VERSION% set JBOSSAS_IP_ADDR=%JBOSSAS_IP_ADDR%"
```
Comment (mmusgrov) [inline scripts/hudson/narayana.bat:17]: NOTBT is referring to not being BlackTie but BlackTie no longer exists, ie it is confusing.

Comment (mmusgrov) [reply scripts/hudson/narayana.bat:17]: I also believe that `NOT [%NOTBT%] == []` is saying if it is BlackTie

File: scripts/hudson/narayana.bat [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  if defined SPI_BRANCH git fetch origin +refs/pull/*/head:refs/remotes/jbosstm/pull/*/head & git checkout %SPI_BRANCH%
+  if %ERRORLEVEL% NEQ 0 exit -1
+  cd ..
+  call build.bat clean install "-f" "jboss-transaction-spi\pom.xml" || (call:comment_on_pull "SPI build Failed %BUILD_URL%" & exit -1)
+goto:eof
+
+:fail_build
+  call:comment_on_pull "Build failed %BUILD_URL%"
+  tasklist & FOR /F "usebackq tokens=5" %%i in (`"netstat -ano|findstr 9990.*LISTENING"`) DO taskkill /F /PID %%i
+  tasklist
```
Comment (mmusgrov) [inline scripts/hudson/narayana.bat:56]: I am reasonably sure that all of the exe's in the tasklist are blacktie executables.

**Review by mmusgrov — CHANGES_REQUESTED** [review]
There are still BlackTie references.

File: scripts/hudson/narayana.bat (hunk 2 above)
Comment (mayankkunwar) [reply scripts/hudson/narayana.bat:17]: okay, I will remove this

File: scripts/hudson/narayana.bat [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+IF NOT [%NOTMAIN%] == [] (call:build_spi_pr
+call build.bat clean install %* || (call:comment_on_pull "Tests failed on Windows - Narayana Failed %BUILD_URL%" & exit -1)
+)
+
+IF NOT [%NOTBT%] == [] (echo "set WILDFLY_MASTER_VERSION=%WILDFLY_MASTER_VERSION% set JBOSSAS_IP_ADDR=%JBOSSAS_IP_ADDR%"
+rem SHUTDOWN ANY PREVIOUS BUILD REMNANTS
+FOR /F "usebackq tokens=5" %%i in (`"netstat -ano|findstr 9990.*LISTENING"`) DO taskkill /F /PID %%i
+tasklist
+taskkill /F /IM mspdbsrv.exe
+taskkill /F /IM testsuite.exe
```
Comment (mayankkunwar) [inline scripts/hudson/narayana.bat:22]: Shall I remove all these exe's in the tasklist?

File: scripts/hudson/narayana.bat (hunk 3 above)
Comment (mayankkunwar) [reply scripts/hudson/narayana.bat:56]: I will remove them, thanks

File: scripts/hudson/narayana.bat [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+if %JAVA_VERSION% == 9 set MAVEN_OPTS="--add-modules=java.corba,java.xml.bind,java.xml.ws"
+
+IF NOT [%NOTMAIN%] == [] (call:build_spi_pr
+call build.bat clean install %* || (call:comment_on_pull "Tests failed on Windows - Narayana Failed %BUILD_URL%" & exit -1)
+)
+
+rem SHUTDOWN ANY PREVIOUS BUILD REMNANTS
+FOR /F "usebackq tokens=5" %%i in (`"netstat -ano|findstr 9990.*LISTENING"`) DO taskkill /F /PID %%i
+tasklist
+wmic Path win32_process Where "Caption Like '%java.exe%' AND CommandLine Like '%standalone%'" Call Terminate
```
Comment (mmusgrov) [inline scripts/hudson/narayana.bat:20]: Isn't this line killing any appserver instances (Like '%standalone%'). I am reasonably sure that the only such app server instances started by this script were the BlackTie ones. But you have already removed the lines in this script which start instances of the app server. Similarly for the other occurrence (the `fail_build` label).

**Review by mmusgrov — CHANGES_REQUESTED** [review]
Suspicious calls to wmic

File: scripts/hudson/narayana.bat (hunk 5 above)
Comment (mayankkunwar) [reply scripts/hudson/narayana.bat:20]: Okay correct, I will remove this

**Review by mmusgrov — APPROVED** [review]
When/if the windows CI run passes this PR will be good to merge, thanks.

**Pull request merged**